activerecord/lib/active_record/middleware/database_selector/resolver/session.rb

```diff
@@ -38,6 +38,9 @@ def last_write_timestamp
           def update_last_write_timestamp
             session[:last_write] = self.class.convert_time_to_timestamp(Time.now)
           end
+
+          def save(response)
+          end
```

Comment:
This could also be removed since this class is really just about the session, and leave `Resolver#update_context(response)` empty. Then, a custom resolver would redefine `update_context` and would use a custom context class that would implement `save(response)` (or whatever it's necessary depending on its `update_context` logic).

Reply:
Is this something you're defining in your app? It feels odd to have an `update_context` method that calls a `save` method that does nothing.

Reply:
Yes, exactly. You're right, it's odd. What do you think of the other alternative I described? That is, not having this method and having `Resolver#update_context` alone instead, which would do nothing as well, but we need to have it so that it can be called from the middleware (or check if `resolver.responds_to?(:update_context)`?)

Reply:
Yea I think having update_context be the empty method makes more sense - that way you don't need to define 2 methods. Something still feels off to me but maybe that's just because I don't completely understand what context is being used for.

Reply:
I get it now! Alright, this makes sense to me. You want to be able to only implement the custom Resolver class and not the rest of it so that's why you implemented the update_context and then save. Sorry for the back and forth ❤️

Reply:
Thank you, and no worries at all! ❤️